### PR TITLE
spacedrive: 0.1.4 -> 0.2.4

### PR DESCRIPTION
**Pull request by DataHearth**
## Description of changes

Diff from `0.1.4` -> `0.2.4`: <https://github.com/spacedriveapp/spacedrive/compare/0.1.4...0.2.4>

## Things done

Update spacedrive project to latest version.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by stepbrobd** [comment]
supersedes and closes #283567
btw 0.2.4 is out

**Comment by DataHearth** [comment]
@stepbrobd Yup, I'll update it right away ^^. So PR is up-to-date

**Comment by DataHearth** [comment]
Package update with latest version (0.2.4)

**Comment by stepbrobd** [comment]
Result of `nixpkgs-review pr 287495` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spacedrive</li>
  </ul>
</details>

**Comment by stepbrobd** [comment]
Result of `nixpkgs-review pr 287495` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spacedrive</li>
  </ul>
</details>

File: pkgs/by-name/sp/spacedrive/package.nix [hunk 1]
```diff
@@ -9,22 +9,23 @@
 
 let
   pname = "spacedrive";
-  version = "0.1.4";
+  version = "0.2.4";
+  name = "${pname}-${version}";
```
Comment (stepbrobd) [inline pkgs/by-name/sp/spacedrive/package.nix:13]: Not needed

See https://github.com/NixOS/nixpkgs/issues/103997

File: pkgs/by-name/sp/spacedrive/package.nix [hunk 2]
```diff
@@ -9,22 +9,23 @@
...
 
   src = fetchurl {
     aarch64-darwin = {
       url = "https://github.com/spacedriveapp/spacedrive/releases/download/${version}/Spacedrive-darwin-aarch64.dmg";
-      hash = "sha256-gKboB5W0vW6ssZHRRivqbVPE0d0FCUdiNCsP0rKKtNo=";
+      sha256 = "ad5466961b2fbc5151af78215f472f7dc24edd695b68d341a3eaf8239e7a6048";
```
Comment (stepbrobd) [inline pkgs/by-name/sp/spacedrive/package.nix:18]: I can't find the exact reference but IIRC `hash` is preferred over `sha256`

File: pkgs/by-name/sp/spacedrive/package.nix [hunk 3]
```diff
@@ -9,22 +9,23 @@
...
-      hash = "sha256-KD1hw6aDyqCsXLYM8WrOTI2AfFx7t++UWV7SaCmtypI=";
+      sha256 = "7ad4407060b94b41b056b0562027c1e5e7fcdf1729ff7e4ad3f42774a98f5121";
     };
     x86_64-linux = {
       url = "https://github.com/spacedriveapp/spacedrive/releases/download/${version}/Spacedrive-linux-x86_64.AppImage";
-      hash = "sha256-iBdW8iPuvztP0L5xLyVs7/K8yFe7kD7QwdTuKJLhB+c=";
+      sha256 = "0fc7ad357ac354b1dad7083eed7bbdc9752f865017c4c5413371a939eac5b2ed";
     };
-  }.${stdenv.system} or (throw "${pname}-${version}: ${stdenv.system} is unsupported.");
+  }.${stdenv.system} or (throw "${name}: ${stdenv.system} is unsupported.");
```
Comment (stepbrobd) [inline pkgs/by-name/sp/spacedrive/package.nix:28]: See the first comment

File: pkgs/by-name/sp/spacedrive/package.nix [hunk 4]
```diff
@@ -33,7 +34,7 @@ let
     platforms = [ "aarch64-darwin" "x86_64-darwin" "x86_64-linux" ];
     license = lib.licenses.agpl3Plus;
     sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
-    maintainers = with lib.maintainers; [ heisfer mikaelfangel stepbrobd ];
+    maintainers = with lib.maintainers; [ heisfer mikaelfangel stepbrobd DataHearth ];
```
Comment (stepbrobd) [inline pkgs/by-name/sp/spacedrive/package.nix:37]: Sort in alphabetical order

**Review by stepbrobd — CHANGES_REQUESTED** [review]

**Review by stepbrobd — APPROVED** [review]

File: pkgs/by-name/sp/spacedrive/package.nix [hunk 5]
```diff
@@ -63,15 +63,16 @@ else appimageTools.wrapType2 {
...
 
       # Install .desktop files
-      install -Dm444 ${appimageContents}/spacedrive.desktop -t $out/share/applications
-      install -Dm444 ${appimageContents}/spacedrive.png -t $out/share/pixmaps
-      substituteInPlace $out/share/applications/spacedrive.desktop \
-        --replace 'Exec=AppRun --no-sandbox %U' 'Exec=spacedrive'
+      install -Dm444 ${appimageContents}/${desktopFile} -t $out/share/applications
+      install -Dm444 ${appimageContents}/${pname}.png -t $out/share/pixmaps
+      substituteInPlace $out/share/applications/${desktopFile} \
+        --replace 'Exec=usr/bin/${pname}' 'Exec=${pname}'
```
Comment (MikaelFangel) [inline pkgs/by-name/sp/spacedrive/package.nix:76]: I'm not sure these changes should be applied. See, https://github.com/NixOS/nixpkgs/pull/270121#discussion_r1405432291

File: pkgs/by-name/sp/spacedrive/package.nix [hunk 6]
```diff
@@ -63,15 +63,16 @@ else appimageTools.wrapType2 {
   extraInstallCommands =
     let
       appimageContents = appimageTools.extractType2 { inherit pname version src; };
+      desktopFile = "com.${pname}.desktop";
     in
     ''
       # Remove version from entrypoint
-      mv $out/bin/spacedrive-"${version}" $out/bin/spacedrive
+      mv $out/bin/${pname}-${version} $out/bin/${pname}
```
Comment (MikaelFangel) [inline pkgs/by-name/sp/spacedrive/package.nix:70]: I'm not sure these changes should be applied. See, https://github.com/NixOS/nixpkgs/pull/270121#discussion_r1405432291

File: pkgs/by-name/sp/spacedrive/package.nix (hunk 5 above)
Comment (DataHearth) [reply pkgs/by-name/sp/spacedrive/package.nix:76]: @MikaelFangel would you prefer having raw naming in those fields ?

**Comment by DataHearth** [comment]
Result of `nixpkgs-review pr 287495` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spacedrive</li>
  </ul>
</details>

**Comment by DataHearth** [comment]
Result of `nixpkgs-review pr 287495` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>spacedrive</li>
  </ul>
</details>

**Pull request merged**